x/rollapp/keeper/msg_server_mark_vulnerable_rollapps.go

```diff
@@ -62,19 +62,31 @@ func (k Keeper) MarkVulnerableRollapps(ctx sdk.Context, drsVersions []string) (i
 			logger.With("rollapp_id", rollapp.RollappId).Info("no latest state info for rollapp")
 			continue
 		}
-		// TODO: this check may be deleted once empty DRS version is marked vulnerable
-		//  https://github.com/dymensionxyz/dymension/issues/1233
-		if info.DrsVersion == "" {
-			logger.With("rollapp_id", rollapp.RollappId).Info("no DRS version set for rollapp")
+
+		// We only check first and last BD to avoid DoS attack related to iterating big number of BDs (taking into account a state update can be submitted with any numblock value)
+		// It is assumed there cannot be two upgrades in the same state update (since it requires gov proposal), if this happens it will be a fraud caught by Rollapp validators.
+		// Therefore checking first and last BD for deprecated DRS version should be enough.
```

Comment:
IMO this is making things a bit more complicated without a good reason
if there is DOS by submitting a batch with loads of BDS it's a pretty serious problem IMO
So either there is a DOS and we should work out what it affect
or there isn't and it's fine
iterating small structs in memory and checking presence in a hashmap is not particularly expensive

Reply:
this has been modified so in only checks the last bd, since the one it matters in the case of marking the rollapp as vulnerable, otherwise it means the rollapp already upgraded and is no longer vulnerable